3-Solving-Problems-By-Searching/bi-directional.js

```diff
@@ -0,0 +1,139 @@
+//Function to clone a object
+function cloneObject(obj) {
+  return JSON.parse(JSON.stringify(obj));
+}
+
+function distance(point1, point2) {
+  return Math.sqrt(Math.pow(point1[0] - point2[0], 2) + Math.pow(point1[1] - point2[1], 2));
+}
+class Node {
+  constructor(id, x, y, adjacent) {
+    this.id = id;
+    this.x = x;
+    this.y = y;
+    this.adjacent = (adjacent != undefined) ? adjacent : [];
+  }
+}
+
+function randomPlanarGraph(height, width, radius) {
+  let [points, grid] = poissonDistribution(height, width, radius);
+
+  let nodes = [];
+  let edges = [];
+  for (let i = 0; i < points.length; i++) {
+    nodes.push(new Node(i, points[i][0], points[i][1]));
+  }
+
+  for (let i = 0; i < nodes.length; i++) {
+    for (let j = i + 1; j < nodes.length; j++) {
+      if (distance(points[i], points[j]) < 2 * radius) {
+        if (Math.random() < 0.9) {
+          nodes[i].adjacent.push(j);
+          nodes[j].adjacent.push(i);
+          edges.push([i, j]);
+        }
+      }
+    }
+  }
+  return [nodes, edges];
+}
+class Graph {
```

Comment:
You've already written a graph class for the other parts of the page; is it possible to reuse that here? Otherwise we will have two incompatible graph classes, two incompatible breadth first searches, etc.